lib/runFirstExistingScript.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+const async = require('async');
+const fs = require('fs');
+const runshell = require('runshell');
+const path = require('path');
+
+// might remove to its own lib for re-use:
+
+const runFirstExistingScript = (fileList, data, options, callback) => {
+  async.detectSeries(fileList, (pathname, detectCallback) => {
+    fs.exists(pathname, (pathExists) => detectCallback(null, pathExists));
+  }, (err, existingScript) => {
+    // if an error:
+    if (err) {
+      return callback(err);
+    }
+    // if none exist:
+    if (!existingScript) {
+      return callback();
+    }
+    if (options.verbose) {
+      options.log(['hubhooks', 'notice'], `running ${existingScript}`);
+    }
+    // todo: does this pass any params to the script?
+    runshell(existingScript, {
+      env: process.env,
```

Comment:
we need to set REPO, BRANCH, USER on env.  I believe new version of runshell already uses process.env